### DISCUSSION TITLE
K3S on Ubuntu?

### DISCUSSION THREAD

**Discussion by aamoroso** [onedr0p/cluster-template#1465] [Q&A] [answered]
Hi,

Wondering if my problem is my choice of Ubuntu over Debian.  I followed the instructions all the way to 

```
task ansible:run playbook=cluster-prepare
```

which completed without error.  When I ran the next step to verify I got an error:

```
❯ kubectl get nodes -o wide
Unable to connect to the server: dial tcp 192.168.0.99:6443: i/o timeout
```

`192.168.0.99` is my `bootstrap_controllers_vip` entry in `config.yaml`

Looking on the nodes there is no k3s service

```
╰─➤  systemctl status k3s.service
Unit k3s.service could not be found.
```

Should it work on Ubuntu?

Thanks!

Adrian

**Comment by onedr0p** [comment]
I would stick with Debian as that is known to work, you're kind of on your own if you want to use Ubuntu.

Debian is a lot better for all this anyways, the amount of bloat that comes with Ubuntu these days is sickening.

However if you want to stick with Ubuntu and figure out your problems feel free to report back!

**Comment by aamoroso** [comment]
Heh, fair enough :)

Thanks for the tip! :)

Cheers,
Adrian

**Comment by aamoroso** [comment, marked as answer]
Ok, it turns out I am an idiot.

I actually didn't run the `cluster-installation` playbook.  I had only run the `cluster-prepare` one.  

I realised when I went looking for the installation role and thought I would just run it manually... then "Wait!  I don't remember that playbook!" 🙄🙄🙄

Anyway, in case anyone is interested, the [ansible-role-k3s](https://github.com/PyratLabs/ansible-role-k3s) role says it is tested on Ubuntu, and it does indeed work.

The playbook failed the first time saying it failed to restart k3s, but on inspection all nodes had the service running.  I ran again and it all worked.  Perhaps 3 retries is not enough for slower hardware? (these are Dell Wyse 5070 boxes).
